test/spec-urls.test.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+const assert = require('assert');
+const specData = require('browser-specs');
+const { visit } = require('../utils');
+
+describe('spec_url data', () => {
+  it('spec_url specifications only use allow listed hosts by w3c/browser-specs (and those we allow list for now)', () => {
+    const toVisit = [
+      'api',
+      'css',
+      'html',
+      'http',
+      'javascript',
+      'mathml',
+      'svg',
+      'webdriver',
+    ];
+
+    let specURLs = [];
+    for (const key of toVisit) {
+      visit(
+        (path, feature) => {
+          if (feature.spec_url) {
+            if (Array.isArray(feature.spec_url)) {
+              specURLs.push(...feature.spec_url);
+            } else {
+              specURLs.push(feature.spec_url);
+            }
+          }
+        },
+        {
+          entryPoint: key,
+        },
+      );
+    }
+
+    let specsNotInBrowserSpecs = [];
```

Comment:
Starting from here, I found the approach to be kinda indirect. I think it'd be a lot more obvious what you're doing if you constructed a single allow list, then checked each URL. Something like this…

```
const specsFromBrowserSpecs = specData.map(
  spec => spec.url || spec.nightly.url || spec.series.nightlyUrl,
);
const specsExceptions = [
// .. long list here ..
]
const allowList = new Set([...specsFromBrowserSpecs, ...specsExceptions]);

const rejectedSpecs = [];
for (const spec of specURLs) {
  // some stuff here that checks whether the spec (minus the hash) is in the set of allowed specs
}
```